### PR TITLE
Update MRMTransitionGroupPicker.h

### PR DESCRIPTION
**Pull request by poshul**
Fixes issues where behavior diverges between architectures due to floating poin weirdness

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by cbielow** [comment]
floating point math should be identical across platforms we support. We are not using fastmath after all.
What problem exactly does this fix?
Introducing magic constants as thresholds does not seem like a robust solution either.

**Comment by poshul** [comment]
> floating point math should be identical across platforms we support.
You'd think, but this is decidedly not so with the apple silicon. Chasing down the test-failures for our new M1 CI I found there are a number of places where the same calculation on M1 returns a very small non-zero number. 
> Introducing magic constants as thresholds does not seem like a robust solution either.
0 is also a magic constant. I do see your point though. I think it might be worth defining a global constant for what we view as the minimum non-zero intensity that makes sense given current instrumentation.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h [hunk 1]
```diff
@@ -233,7 +233,7 @@ namespace OpenMS
       OPENMS_PRECONDITION(transition_group.chromatogramIdsMatch(), "Chromatogram native IDs need to match keys in transition group")
 
       MRMFeature mrmFeature;
-      mrmFeature.setIntensity(0.0);
+      mrmFeature.setIntensity();
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h:236]:
```suggestion
      mrmFeature.setIntensity(0.0);
```

Comment (jpfeuffer) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h]: also the accumulate changes will not compile

**Comment by jpfeuffer** [comment]
I don't think floating point precision can be the same on Silicon.
Silicon does not have 80-bit registers for internal computations.

**Comment by timosachsenberg** [comment]
K let’s merge but keep these differences in mind.

**Comment by timosachsenberg** [comment]
@poshul is this ready to go from your side?

**Comment by cbielow** [comment]
since you merged this now, noone will ever look at this again, I'm afraid.

**Comment by cbielow** [comment]
> @cbielow how would you debug this?

Lots of printf with a linux box vs. an M1/M2. Should not take long, but requires access to an M1...

**Comment by timosachsenberg** [comment]
> since you merged this now, noone will ever look at this again, I'm afraid.

I can open an issue, but not sure if this increases the chance that someone looks at it again. Or the other numerical differences not being tested ;)

**Comment by poshul** [comment]
I've opened a bug report for this to keep it alive.

**Comment by jpfeuffer** [comment]
Which tests were failing for this issue?

**Comment by jpfeuffer** [comment]
Cannot reproduce.. `ctest -R MRM` does not fail

**Comment by timosachsenberg** [comment]
on M1?

**Comment by jpfeuffer** [comment]
yes

**Comment by timosachsenberg** [comment]
@poshul can you shed light into this?

**Comment by poshul** [comment]
None of the tests failed after this change, there was one that failed *before* on the M1.

**Comment by jpfeuffer** [comment]
Sure. I reverted the changes. Which one?

**Comment by poshul** [comment]
> Sure. I reverted the changes. Which one?

Sorry, what? I feel like we are talking past each other here.

**Comment by jpfeuffer** [comment]
I reverted your changes and built it on abibuilder9. No test failures. Which ones did you see and try to fix with this PR?

**Pull request merged**